common/lib/dependabot/shared_helpers.rb

```diff
@@ -37,10 +37,12 @@ def initialize(error_class:, error_message:, error_backtrace:)
 
     def self.in_a_temporary_repo_directory(directory = "/",
                                            repo_contents_path = nil,
+                                           branch = nil,
                                            &block)
       if repo_contents_path
         path = Pathname.new(File.join(repo_contents_path, directory)).
                expand_path
+        checkout_git_branch(repo_contents_path, branch) if branch
```

Comment:
☕ 🚗  will this not no-op with the current implementation of `checkout_git_branch`? https://github.com/dependabot/dependabot-core/blob/14ddb5b533f72a0b31c2398ef2c4ea2a5d9bf9a6/common/lib/dependabot/file_fetchers/base.rb#L436

Reply:
I ran into this in some tests and while dry running, but those tests have since been changed to remove the cloned repo, and in a prod environment the repo should never be left behind, so you're right 👍